js/main.js

```diff
@@ -25,28 +23,59 @@ function handleClick(event) {
     swapToSearchView();
     unfillStar();
   } else if (event.target.matches('.fa-regular')) {
+    var $parentAppend = document.querySelector('.row.bg-grey');
     fillStar();
     saveCard();
+    var recentFav = renderElement(data.savedCards[0]);
+    $parentAppend.prepend(recentFav);
+  } else if (event.target.matches('.favorite-anchor') || event.target.matches('.favorite-button')) {
+    swapToFavoriteView();
+  } else if (event.target.matches('.favorite-image')) {
+    showModal();
+    var $targetDiv = event.target.closest('[data-card-id]');
+    var cardID = parseInt($targetDiv.getAttribute('data-card-id'));
+    for (var i = 0; i < data.savedCards.length; i++) {
+      if (cardID === data.savedCards[i].savedCardID) {
+        data.inspectedCard = data.savedCards[i];
+      }
+    }
+    var chosenCard = data.inspectedCard;
+    repopulateModal(chosenCard);
+  } else if (event.target.matches('.modal-content') || (event.target.matches('.img-container'))) {
+    hideModal();
   }
```

Comment:
This function is doing a LOT of actions. Instead of delegating all of this from the `document`, I would delegate for each of these events in separate event handlers placed on the immediate ancestor of the elements you are trying to receive the event from.